### PR TITLE
Add maven publish for 1.3.x release

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Add maven publish for 1.3.x release

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/3331

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch/maven-publish-1.3.x.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,90 @@
+lib = library(identifier: 'jenkins@1.5.4', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/opensearch-project/opensearch-build-libraries.git',
+]))
+
+pipeline {
+    options {
+        timeout(time: 4, unit: 'HOURS')
```
Comment (gaiksaya) [inline jenkins/opensearch/maven-publish-1.3.x.jenkinsfile:8]: 2 hours should be more than enough

File: jenkins/opensearch/maven-publish-1.3.x.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            }
+            steps {
+                script {
+                    echo("Detect Docker Images and Related Parameters")
+                    dockerAgent = detectDockerAgent()
+                    currentBuild.description = INPUT_MANIFEST
+                }
+            }
+        }
+        stage('build-distribution-snapshot-linux-x64-tar') {
```
Comment (gaiksaya) [inline jenkins/opensearch/maven-publish-1.3.x.jenkinsfile:49]:
```suggestion
        stage('publish-maven-snaphots') {
```

File: jenkins/opensearch/maven-publish-1.3.x.jenkinsfile (hunk 1 above)
Comment (prudhvigodithi) [reply jenkins/opensearch/maven-publish-1.3.x.jenkinsfile]: Updated Sayali

File: jenkins/opensearch/maven-publish-1.3.x.jenkinsfile (hunk 2 above)
Comment (prudhvigodithi) [reply jenkins/opensearch/maven-publish-1.3.x.jenkinsfile]: Updated Sayali

File: jenkins/opensearch/maven-publish-1.3.x.jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                        platform: 'linux',
+                        architecture: 'x64',
+                        distribution: 'tar',
+                        snapshot: true
+                    )
+
+                    String mavenPath = "$WORKSPACE/tar/builds/opensearch/maven"
+
+                    if (fileExists(mavenPath)) {
+                        withCredentials([usernamePassword(credentialsId: 'jenkins-sonatype-creds', usernameVariable: 'SONATYPE_USERNAME', passwordVariable: 'SONATYPE_PASSWORD')]) {
```
Comment (gaiksaya) [inline jenkins/opensearch/maven-publish-1.3.x.jenkinsfile:75]: Can we use https://github.com/opensearch-project/common-utils/blob/main/.github/workflows/maven-publish.yml#L34-L35 credentials for snapshots?

Comment (prudhvigodithi) [reply jenkins/opensearch/maven-publish-1.3.x.jenkinsfile]: Updated Sayali

**Review by jordarlu — APPROVED** [review]

**Pull request merged**